extensions/amp-story/1.0/page-advancement.js

```diff
@@ -721,10 +733,57 @@ class MediaBasedAdvancement extends AdvancementConfig {
     /** @private {?../../../src/video-interface.VideoInterface} */
     this.video_ = null;
 
-    if (element.ownerDocument.defaultView) {
-      /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
-      this.storeService_ =
-        getStoreService(element.ownerDocument.defaultView);
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = getStoreService(win);
+
+    this.elements_.forEach(el => {
+      listen(el, VideoEvents.VISIBILITY, () => this.onVideoVisibilityChange_());
+    });
+  }
+
+  /**
+   * Returns the first playable element, or null. An element is considered
+   * playable if it's either visible, or a hidden AMP-AUDIO.
+   * @return {?Element}
+   * @private
+   */
+  getFirstPlayableElement_() {
+    if (this.elements_.length === 1) {
+      return this.elements_[0];
+    }
+
+    for (let i = 0; i < this.elements_.length; i++) {
+      const element = this.elements_[i];
+      const resource = this.resources_.getResourceForElement(element);
+      if (resource.isDisplayed() ||
+          (element.tagName === 'AMP-AUDIO' &&
```

Comment:
is it worth trying to actually use a common function for the condition both used here and in `amp-story-page`'s `isMediaDisplayed_`?

Reply:
Done